extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx

```diff
@@ -33,6 +34,22 @@ export default createHigherOrderComponent(
 
 		const [ isVisible, setIsVisible ] = useState( ! isDualMode );
 
+		const bannerContext = 'editor-canvas';
+
+		const trackEventData = {
+			plan: requiredPlan,
+			blockName: props.name,
+			context: bannerContext,
+		};
+
+		useEffect( () => {
+			if ( ! isVisible ) {
+				return;
+			}
+
+			trackUpgradeBannerImpression( trackEventData );
+		}, [ isVisible, trackEventData ] );
```

Comment:
It seems that the event `jetpack_editor_block_upgrade_banner_impression` is fired also when clicking outside the block. Also it is fired on page reload without the banner being shown. Using `isBannerVisible` as dependency instead of `isVisible` seems to resolve that issue for me.

As a note also, It seems that the event `jetpack_editor_block_upgrade_banner_impression` is fired every-time the banner is visible. So clicking outside the block and then inside triggers the event. I believe we should only track the event once? Can't see the value of tracking the banner impression many times in such a short period of time but maybe I am missing something. cc @davemart-in who maybe can enlighten us.

![SS 2020-08-19 at 15 10 11](https://user-images.githubusercontent.com/12430020/90633181-43acea00-e22e-11ea-9088-17408119888c.gif)

Reply:
Nice 👀 , Harris as usual 👏 

`IsBannerVisible` has been addressed [here](https://github.com/Automattic/jetpack/pull/16882/commits/70fa5e00c2f0addc323a1f84715f46cbe370f2b8).

Asking to @lessbloat: should we record the `jetpack_editor_block_upgrade_banner_impression` track-event every time that the banner shows up, or just once, in the first time.
The old Upgrade Nudge was visible constantly so it used to record the event just once.

Reply:
Just once. Thanks!

Reply:
Addressed [here](33a9bde9cb974111fe3f39164612cda15bd5f407)